core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/Eth2SignForIdentifierHandler.java

```diff
@@ -71,8 +69,12 @@ public void handle(final RoutingContext routingContext) {
           .ifPresentOrElse(
               signature -> {
                 if (slashingProtection.isPresent()) {
-                  if (maySign(identifier, eth2SigningRequestBody)) {
-                    respondWithSignature(routingContext, signature);
+                  try {
+                    if (maySign(identifier, eth2SigningRequestBody)) {
+                      respondWithSignature(routingContext, signature);
+                    }
```

Comment:
should there be an "else" here? responding with appropriate `routingContext.fail(code)` ??

Reply:
done